sql/core/src/test/scala/org/apache/spark/sql/DataFrameSetOperationsSuite.scala

```diff
@@ -506,4 +506,23 @@ class DataFrameSetOperationsSuite extends QueryTest with SharedSparkSession {
     check(lit(2).cast("int"), $"c" === 2, Seq(Row(1, 1, 2, 0), Row(1, 1, 2, 1)))
     check(lit(2).cast("int"), $"c" =!= 2, Seq())
   }
+
+  test("SPARK-29358: Make unionByName optionally fill missing columns with nulls") {
+    withSQLConf(SQLConf.ALLOW_MISSING_COLUMNS_IN_UNION_BY_NAME.key -> "true") {
+      var df1 = Seq(1, 2, 3).toDF("a")
+      var df2 = Seq(3, 1, 2).toDF("b")
+      val df3 = Seq(2, 3, 1).toDF("c")
+      val unionDf = df1.unionByName(df2.unionByName(df3))
```

Comment:
We need to union the case with no common column? The behaivour looks like `mergeSchema` in parquet.

Reply:
This shows the behavior of enabling this config. Yeah, it looks like to merge the schema of different datasets.